pkg/asset/installconfig/powervs/session.go

```diff
@@ -33,62 +57,162 @@ type PISessionVars struct {
 	Zone   string `json:"zone,omitempty"`
 }
 
-// GetSession returns an ibmpisession object.
-func GetSession() (*Session, error) {
-	s, apiKey, err := getPISession()
+func authenticateAPIKey(sess *bxsession.Session) error {
+	config := sess.Config
+	tokenRefresher, err := authentication.NewIAMAuthRepository(config, &rest.Client{
+		DefaultHeader: gohttp.Header{
+			"User-Agent": []string{http.UserAgent()},
+		},
+	})
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to load credentials")
+		return err
 	}
-	return &Session{Session: s, APIKey: apiKey}, nil
+	return tokenRefresher.AuthenticateAPIKey(config.BluemixAPIKey)
 }
 
-var (
-	defaultAuthFilePath = filepath.Join(os.Getenv("HOME"), ".powervs", "config.json")
-)
+func fetchUserDetails(sess *bxsession.Session) (*User, error) {
+	config := sess.Config
+	user := User{}
+	var bluemixToken string
 
-func getPISession() (*ibmpisession.IBMPISession, string, error) {
+	if strings.HasPrefix(config.IAMAccessToken, "Bearer") {
+		bluemixToken = config.IAMAccessToken[7:len(config.IAMAccessToken)]
+	} else {
+		bluemixToken = config.IAMAccessToken
+	}
 
-	var err error
-	var pisv PISessionVars
+	token, err := jwt.Parse(bluemixToken, func(token *jwt.Token) (interface{}, error) {
+		return "", nil
+	})
+	if err != nil && !strings.Contains(err.Error(), "key is of invalid type") {
+		return &user, err
+	}
+
+	claims := token.Claims.(jwt.MapClaims)
+	if email, ok := claims["email"]; ok {
+		user.Email = email.(string)
+	}
+	user.ID = claims["id"].(string)
+	user.Account = claims["account"].(map[string]interface{})["bss"].(string)
+
+	return &user, nil
+}
 
-	// Grab variables from the installer written authFilePath.
-	err = getPISessionVarsFromAuthFile(&pisv)
+//NewBxClient func returns bluemix client
+func NewBxClient() (*BxClient, error) {
+	c := &BxClient{}
+
+	var pisv PISessionVars
+	// Grab variables from the installer written authFilePath
+	logrus.Debug("Gathering variables from AuthFile")
+	err := getPISessionVarsFromAuthFile(&pisv)
 	if err != nil {
-		return nil, "", err
+		return nil, err
 	}
 
-	// Grab variables from the users environment.
+	// Grab variables from the users environment
+	logrus.Debug("Gathering variables from user environment")
 	err = getPISessionVarsFromEnv(&pisv)
 	if err != nil {
-		return nil, "", err
+		return nil, err
 	}
 
 	// Prompt the user for the remaining variables.
 	err = getPISessionVarsFromUser(&pisv)
 	if err != nil {
-		return nil, "", err
+		return nil, err
 	}
 
 	// Save variables to disk.
 	err = savePISessionVars(&pisv)
 	if err != nil {
-		return nil, "", err
+		return nil, err
 	}
 
-	piOpts := ibmpisession.IBMPIOptions{
-		Authenticator: &core.IamAuthenticator{
-			ApiKey: pisv.APIKey,
-		},
-		Region:      pisv.Region,
-		UserAccount: pisv.ID,
-		Zone:        pisv.Zone,
+	c.APIKey = pisv.APIKey
+
+	os.Setenv("IC_API_KEY", pisv.APIKey)
```

Comment:
I think we should refrain from setting environment variables since you cannot rely on them. Do we need this? I see only one place in the code where this is used and it looks like savePISessionVars function should take care of it?

Reply:
updated the PR with changes